_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1039 @@
+
+## Tutorial overview
+
+This tutorial consists of the following steps:
+
+1. [**Set up an ML language model**](#step-1-set-up-an-ml-language-model)
+    1. [Choose a language model](#step-1a-choose-a-language-model)
+    1. [Register a model group](#step-1b-register-a-model-group)
+    1. [Register the model to the model group](#step-1c-register-the-model-to-the-model-group)
+    1. [Deploy the model](#step-1d-deploy-the-model)
+1. [**Ingest data with neural search**](#step-2-ingest-data-with-neural-search)
+    1. [Create an ingest pipeline for neural search](#step-2a-create-an-ingest-pipeline-for-neural-search)
+    1. [Create a k-NN index](#step-2b-create-a-k-nn-index)
+    1. [Ingest documents into the index](#step-2c-ingest-documents-into-the-index)
+1. [**Search the data**](#step-3-search-the-data)
+   - [Search with a keyword search](#search-with-a-keyword-search)
+   - [Search with a neural search](#search-with-a-neural-search)
+   - [Search with a combined keyword search and neural search](#search-with-a-combined-keyword-search-and-neural-search)
+
+Some steps in the tutorial contain optional `Test it` sections. You can ensure the step worked by running requests in these sections.
+
+After you're done, follow the steps in the [Clean up](#clean-up) section to delete all created components.
+
+## Tutorial
+
+You can follow this tutorial using your command line or the OpenSearch Dashboards [Dev Tools console]({{site.url}}{{site.baseurl}}/dashboards/dev-tools/run-queries/).
+
+## Step 1: Set up an ML language model
+
+Neural search requires a language model to generate vector embeddings from text fields both at ingestion time and query time.
+
+### Step 1(a): Choose a language model
+
+For this tutorial, you'll use the [DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) model from Hugging Face. It is one of the pretrained sentence transformer models available in OpenSearch. You'll need the name, version, and dimension of the model to register it. You can find this information in the [pretrained model table]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers) by selecting the `config_url` link for the TorchScript artifact of the model:
+
+- The model name is `huggingface/sentence-transformers/msmarco-distilbert-base-tas-b`.
+- The model version is `1.0.1`.
+- The number of dimensions for this model is `768`.
+
+#### Advanced: Using a different model
+
+You can choose to use another model:
+
+- One of the [pretrained language models provided by OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/).
+- Your own model. For instructions how to set up a custom model, see [Model serving framework]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-framework/).
```

Comment:
Should we change `Model serving framework` to `ML framework`?
```suggestion
- Your own model. For instructions how to set up a custom model, see [ML framework]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-framework/).
```

Reply:
Already changed, thanks!